### PR TITLE
ForumPageDiscover - Created connection between hashtags API and forum page discover

### PR DESCRIPTION
**Pull request by Bcastillo44**
- Related Issue (include '#'): #672 

- Description of changes made:

Created connection hashtags from API to hashtags on Forum Page Discover

- Is the feature complete/bug resolved/etc..:

Need to add pagination

### CODE REVIEW

File: client/src/pages/ForumPage/ForumPageDiscover.js [hunk 1]
```diff
@@ -20,7 +23,32 @@ const handleClick = e => {
...
+    .then(response => {
+      this.setState({ 
+        hashtags: response.data
+      })
+      console.log(response.data);
+    })
+    .catch(error => {
+      console.log('Error', error);
+    })
+}
```
Comment (0xZakk) [inline client/src/pages/ForumPage/ForumPageDiscover.js]: Can you put this in the appropriate service in the `services/` directory?

File: client/src/pages/ForumPage/ForumPageDiscover.js [hunk 2]
```diff
@@ -20,7 +23,32 @@ const handleClick = e => {
...
+    }
+  }
+
+  componentDidMount() {
+  axios.get(`${BASE_URL}/hashtags`)
+    .then(response => {
+      this.setState({ 
+        hashtags: response.data
+      })
+      console.log(response.data);
```
Comment (0xZakk) [inline client/src/pages/ForumPage/ForumPageDiscover.js]: Can you remove this and any other `console.log` statements?

**Review by 0xZakk — CHANGES_REQUESTED** [review]
@Bcastillo44

File: client/src/pages/ForumPage/ForumPageDiscover.js (hunk 2 above)
Comment (Bcastillo44) [reply client/src/pages/ForumPage/ForumPageDiscover.js]: Removed console.logs from the page

File: client/src/pages/ForumPage/ForumPageDiscover.js (hunk 1 above)
Comment (Bcastillo44) [reply client/src/pages/ForumPage/ForumPageDiscover.js]: Added Forum.js page in services/ directory and created import/export of axios fetch

File: client/src/pages/ForumPage/ForumPageDiscover.js [hunk 3]
```diff
@@ -9,18 +9,38 @@ import Button from "../../components/Button/Button";
...
+    this.getHashtags = getHashtags.bind(this);
+  }
+
+  componentDidMount() {
+  this.getHashtags()
+}
+
+render() {
+
+  const hashtags = this.state.hashtags ? this.state.hashtags: [];
```
Comment (0xZakk) [inline client/src/pages/ForumPage/ForumPageDiscover.js]: The formatting here is really weird - can you make sure you have Prettier installed and set up to auto format on paste and on save?

**Review by 0xZakk — CHANGES_REQUESTED** [review]
@Bcastillo44

File: client/src/pages/ForumPage/ForumPageDiscover.js (hunk 3 above)
Comment (Bcastillo44) [reply client/src/pages/ForumPage/ForumPageDiscover.js]: Double-checked and I do... manually changed the formatting

**Comment by Bcastillo44** [comment]
I cannot seem to get those text-decoration on the pills to be fixed. I tired every CSS on the page, and nothing would resolve the issue. Not sure how to proceed, or if we can just keep it on there for now.

**Comment by Bcastillo44** [comment]
Found the fix for the styling!

File: client/src/pages/ForumPage/ForumPage.css [hunk 4]
```diff
@@ -38,6 +38,10 @@
   margin: 10px 50px 30px 0px;
 }
 
+a:link {
+  text-decoration: none!important;
+}
```
Comment (0xZakk) [inline client/src/pages/ForumPage/ForumPage.css]: Can you use the css naming convention we discussed? Also, you'll need to find a way to fix this without using `!important`. You have some styles clashing somewhere. Can you debug that?

**Review by 0xZakk — CHANGES_REQUESTED** [review]
@Bcastillo44 

See my comment. Something is also still off with the formatting of your code. Can you figure out what's going on and reformat these files?

File: client/src/pages/ForumPage/ForumPage.css (hunk 4 above)
Comment (Bcastillo44) [reply client/src/pages/ForumPage/ForumPage.css]: Updated CSS with inline text-decoration fix

File: client/src/pages/ForumPage/ForumPageDiscover.js [hunk 5]
```diff
@@ -9,247 +9,166 @@ import Button from "../../components/Button/Button";
...
-            color={"pink"}
-            icon={"pink"}
-            shadow
-            border
-            onClick={pillClick}
-            selectId={0}
-          />
+
+        <div className="forum-page-body">
+          <div className="forum-page-sort">
```
Comment (0xZakk) [inline client/src/pages/ForumPage/ForumPageDiscover.js]: Can you use the CSS naming convention from the readme? For all your CSS classes!

File: client/src/pages/ForumPage/ForumPageDiscover.js [hunk 6]
```diff
@@ -9,247 +9,166 @@ import Button from "../../components/Button/Button";
...
+          <div>
+            <header className="ForumPage__header">
+              #hashtags: <a href="/forum-page-hashtag">Example Tag</a>
+            </header>
+          </div>
+          <div className="forum-page-hollow-pill-container">
+            {hashtags.map(hashtag => {
+              return (
+                <div className="Forum-Pill">
+                  <Link to={`/forum-page-hashtag/${hashtag.id}`} style={{ textDecoration: "none" }}>
```
Comment (0xZakk) [inline client/src/pages/ForumPage/ForumPageDiscover.js]: The page this links to is blank. Can we have it link to the sample page?

File: client/src/pages/ForumPage/ForumPageDiscover.js [hunk 7]
```diff
@@ -9,247 +9,166 @@ import Button from "../../components/Button/Button";
...
+              size={"short"}
+              color={"pink"}
+              icon={"white"}
+              onClick={pillClick}
+              selectId={0}
+            />
+          </div>
+          <div>
+            <header className="ForumPage__header">
+              #hashtags: <a href="/forum-page-hashtag">Example Tag</a>
```
Comment (0xZakk) [inline client/src/pages/ForumPage/ForumPageDiscover.js]: Can you remove the link to the example page?

File: client/src/pages/ForumPage/ForumPageDiscover.js [hunk 8]
```diff
@@ -9,247 +9,166 @@ import Button from "../../components/Button/Button";
...
+                      border
+                      onClick={pillClick}
+                      selectId={hashtag.id}
+                    />
+                  </Link>
+                </div>
+              );
+            })}
+          </div>{" "}
+          <div className="ForumPage__button-container"></div>
```
Comment (0xZakk) [inline client/src/pages/ForumPage/ForumPageDiscover.js:166]: Can you remove this empty div?

**Review by 0xZakk — CHANGES_REQUESTED** [review]
@Bcastillo44

File: client/src/pages/ForumPage/ForumPageDiscover.js (hunk 5 above)
Comment (Bcastillo44) [reply client/src/pages/ForumPage/ForumPageDiscover.js]: I did not create these class names... these were the names given on the page when I started working on these pages... but I can fix them...

File: client/src/pages/ForumPage/ForumPageDiscover.js (hunk 6 above)
Comment (Bcastillo44) [reply client/src/pages/ForumPage/ForumPageDiscover.js]: Yes, I can change the link to the sample page. I think Chris wanted them to go to the appropriate hashtags that he created, but I can change it to the sample.

File: client/src/pages/ForumPage/ForumPageDiscover.js (hunk 7 above)
Comment (Bcastillo44) [reply client/src/pages/ForumPage/ForumPageDiscover.js]: Link removed.

File: client/src/pages/ForumPage/ForumPageDiscover.js (hunk 8 above)
Comment (Bcastillo44) [reply client/src/pages/ForumPage/ForumPageDiscover.js:166]: Empty div removed.

File: client/src/pages/ForumPage/ForumPageDiscover.js (hunk 5 above)
Comment (0xZakk) [reply client/src/pages/ForumPage/ForumPageDiscover.js]: Thanks! Boy scout rule (leave it better than you found it)

**Review by 0xZakk — CHANGES_REQUESTED** [review]
Getting this error when I try to load the hashtags?
```sh
Internal Server Error: /hashtags
Traceback (most recent call last):
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.UndefinedColumn: column forum_post_topics.hashtag_id does not exist
LINE 1: ...forum_post_topics" ON ("community_hashtag"."id" = "forum_pos...
                                                             ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/views.py", line 505, in dispatch
    response = self.handle_exception(exc)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/views.py", line 465, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/views.py", line 476, in raise_uncaught_exception
    raise exc
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/views.py", line 502, in dispatch
    response = handler(request, *args, **kwargs)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/generics.py", line 239, in get
    return self.list(request, *args, **kwargs)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/mixins.py", line 46, in list
    return Response(serializer.data)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/serializers.py", line 760, in data
    ret = super().data
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/serializers.py", line 260, in data
    self._data = self.to_representation(self.instance)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/serializers.py", line 677, in to_representation
    return [
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/serializers.py", line 678, in <listcomp>
    self.child.to_representation(item) for item in iterable
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/serializers.py", line 529, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/serializers.py", line 677, in to_representation
    return [
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/serializers.py", line 678, in <listcomp>
    self.child.to_representation(item) for item in iterable
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/serializers.py", line 529, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/rest_framework/relations.py", line 533, in to_representation
    return [
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/db/models/query.py", line 276, in __iter__
    self._fetch_all()
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/db/models/query.py", line 1261, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/db/models/query.py", line 57, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/db/models/sql/compiler.py", line 1152, in execute_sql
    cursor.execute(sql, params)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/db/backends/utils.py", line 100, in execute
    return super().execute(sql, params)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/db/backends/utils.py", line 68, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/db/backends/utils.py", line 77, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/Users/zacharyfleischmann/.local/share/virtualenvs/api-YO8aAAYX/lib/python3.8/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: column forum_post_topics.hashtag_id does not exist
LINE 1: ...forum_post_topics" ON ("community_hashtag"."id" = "forum_pos...
                                                             ^
```

**Comment by Bcastillo44** [comment]
I had to remigrate the API

**Review by 0xZakk — CHANGES_REQUESTED** [review]
I already tried remigrating. This is what I get: 

```sh
cotripper/api on  forum-page-discover-hashtags-api-connect [$] via 🐍 v3.8.3 (api-YO8aAAYX) on ☁️  us-east-1
❯ python manage.py migrate --settings api.settings.local
System check identified some issues:

WARNINGS:
accounts.Profile.activities: (fields.W340) null has no effect on ManyToManyField.
accounts.Profile.connections: (fields.W340) null has no effect on ManyToManyField.
accounts.Profile.events: (fields.W340) null has no effect on ManyToManyField.
accounts.Profile.groups: (fields.W340) null has no effect on ManyToManyField.
accounts.Profile.hashtags: (fields.W340) null has no effect on ManyToManyField.
accounts.Profile.topics: (fields.W340) null has no effect on ManyToManyField.
forum.Post.hashtags: (fields.W340) null has no effect on ManyToManyField.
forum.Post.parent: (fields.W340) null has no effect on ManyToManyField.
forum.Post.topics: (fields.W340) null has no effect on ManyToManyField.
Operations to perform:
  Apply all migrations: accounts, admin, auth, community, contenttypes, cotrip_app, forum, knox, sessions, trip
Running migrations:
  No migrations to apply.
```

**Comment by Bcastillo44** [comment]
I re-loaded the page and this is what I currently am seeing:


Screen Shot 2020-07-14 at 2 15 20 PM

**Comment by Bcastillo44** [comment]
This has been updated with the correct CSS

**Comment by secondaryfun** [comment]
Branch https://github.com/CotripperPlatform/CoTrip/pull/685 has these changes. This branch can be closed without merging.

**Pull request closed without merging**